students/DanCornutt/session08/circle.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+"""Circle HW, Session 07
+
+"""
+from math import pi
+
+class Circle:
+    """class for Circles with attributes and methods used by all."""
+    name = "Circle"
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        """Returns diameter of circle."""
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        """Sets diameter of constructed circle.
+        param1: diameter of wanted circle"""
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        """Computes area of circle. Returns area"""
+        return pi * self.radius ** 2
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        """Alternate constructer for from_diameter
+        param1 : diameter of wanted circle"""
+        return cls(diameter / 2)
+
+
+    def __str__(self):
+        return "{} with radius {:.6f}".format(self.name, self.radius)
```

Comment:
ahh! got it -- using name attribute sothat subclasses will work with the same __str__.

you can get that with metaprogramming trickery:

```
In [61]: class C: 
    ...:     def test_name(self): 
    ...:         print("class name is:") 
    ...:         print(self.__class__.__name__) 
    ...:                                                                        

In [62]: c = C()                                                                

In [63]: c.test_name()                                                          
class name is:
C
```

we were not expecting you to figure that out at this point :-)